packages/console/src/components/UserName/index.tsx

```diff
@@ -18,18 +22,32 @@ type Props = {
 };
 
 function UserName({ userId, isLink = false }: Props) {
-  const { data, error } = useSWR<User, RequestError>(`api/users/${userId}`);
-  const isLoading = !data && !error;
-  const name = conditionalString(data && getUserTitle(data));
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const fetchApi = useApi({ hideErrorToast: true });
+  const fetcher = useSwrFetcher<User>(fetchApi);
+  const { data, isLoading, error } = useSWR<User, RequestError>(`api/users/${userId}`, {
+    fetcher,
+    shouldRetryOnError: shouldRetryOnError({ ignore: [404] }),
+  });
   const { getTo } = useTenantPathname();
 
+  const name = useMemo(() => {
+    if (data) {
+      return getUserTitle(data);
+    }
+    if (error?.status === 404) {
+      return `${userId} (${t('general.deleted')})`;
+    }
+    return '-';
```

Comment:
nit: `N/A`?

Reply:
The `-` is consistent with other usages in Console. So I think we'll just keep it as-is for now.